pallets/loans/src/lib.rs

```diff
@@ -398,12 +423,15 @@ pub mod pallet {
 			amount: T::Balance,
 		) -> DispatchResultWithPostInfo {
 			let owner = ensure_signed(origin)?;
-			let first_borrow = Self::borrow_amount(pool_id, loan_id, owner, amount)?;
+			let (active_count, first_borrow) =
+				Self::borrow_amount(pool_id, loan_id, owner, amount)?;
 			Self::deposit_event(Event::<T>::Borrowed(pool_id, loan_id, amount));
-			match first_borrow {
-				true => Ok(Some(T::WeightInfo::initial_borrow()).into()),
-				false => Ok(Some(T::WeightInfo::further_borrows()).into()),
-			}
+
+			Ok(if first_borrow {
```

Comment:
Not a fan of this syntax and `clippy`, which we would like to introduce some time soonish, will complain as well.
My preference here would be an `if first_borrow { ... } else { ... }` or just keep the same `match` approach used before.

Reply:
Pretty sure clippy complains about match on bools. I can pull the `Ok` into the if, I was just being clever

Reply:
I ended up going a slightly different way with this, but clippy seems to be happy. What do you think?